### DISCUSSION TITLE
Add button in admin to revert changes.

### DISCUSSION THREAD

**Discussion by aywein** [AmbitionEng/django-pghistory#160] [Ideas]
Thank you for this amazing project!
I'm crying for all year that I never knew it...

It would be useful to have a button in the admin UI at the objects event page to revert this event.
I know you have this feature in code already. But it would be good to have it in admin as well.

Thank you!

**Comment by wesleykendall** [comment]
@aywein thanks for the kind words! I converted this to a discussion for now.

Are you thinking that the [events admin](https://django-pghistory.readthedocs.io/en/stable/admin/#the-events-admin) would have check boxes next to the events and an action to revert? Or are you expecting to click into an individual event and then hit "revert"?

I'm assuming it would be the latter. Event models that don't track all fields also wouldn't be able to be reverted, so I'd assume the button would be disabled for those.

 I'd welcome a PR that adds reversion functionality to the admin if anyone wants to try. My top priority with pghistory at the moment is https://github.com/Opus10/django-pghistory/discussions/158, but I could imagine prioritizing this after if no progress has been made by anyone else

**Reply by aywein** [reply]
Both would be helpful. the first one is when you updated mistakenly a few items, the second option is perfect for reverting a specific change.

Note that this can include not just reverting one step back, but even reverting to any point in history by opening that old event and clicking revert to this version.

Thank you.

**Reply by qqii** [reply]
+1

`django-simple-history`'s implementation is what such a feature may look like: https://django-simple-history.readthedocs.io/en/stable/admin.html
